src/views/v3/Bridge/Routes/index.tsx

```diff
@@ -51,16 +54,93 @@ function Routes({
   const [originalSelectedRoute, setOriginalSelectedRoute] = useState<
     string | undefined
   >(selectedRoute);
+  const [selectedRouteBadge, setSelectedRouteBadge] =
+    useState<string>('fastest');
+
+  const routesWithQuotes = routesList.filter((rs) => quotes[rs] !== undefined);
+
+  const selectedQuote = !selectedRoute
+    ? undefined
+    : (() => {
+        const quoteResult = quotes[selectedRoute];
+        return quoteResult?.success ? quoteResult : undefined;
+      })();
+
+  const fastestRoute = routesWithQuotes.reduce(
+    (fastest, route) => {
+      const quote = quotes[route];
+      if (!quote || !quote.success) return fastest;
+
+      if (quote.eta !== undefined && quote.eta < fastest.eta) {
+        return { name: route, eta: quote.eta };
+      }
+      return fastest;
+    },
+    { name: '', eta: Infinity },
+  );
+
+  const cheapestRoute = routesWithQuotes.reduce(
+    (cheapest, route) => {
+      const quote = quotes[route];
+      const rc = config.routes.get(route);
+      if (!quote || !quote.success || !rc.AUTOMATIC_DEPOSIT) return cheapest;
+
+      const amountOut = BigInt(quote.destinationToken.amount.amount);
+      if (amountOut > cheapest.amountOut) {
+        return { name: route, amountOut };
+      }
+      return cheapest;
+    },
+    { name: '', amountOut: 0n },
+  );
 
   useEffect(() => {
     // Reset the highlighted route when the selected route changes
-    if (selectedRoute && selectedRoute !== highlightedRoute) {
-      setHighlightedRoute(selectedRoute);
+    if (selectedRoute) {
+      if (selectedRoute !== highlightedRoute) {
+        setHighlightedRoute(selectedRoute);
+      }
+
+      // Set the selected route badge based on the selected route
+      if (selectedRoute === fastestRoute.name) {
```

Comment:
onChange below calls onRouteChange, does that send in the new selectedRoute to this component?

If so, you might be able to simplify quite a bit and reduce extra renders as you wont need state for the badge.  As in, selectedRoute is probably already in some state somewhere and everything else can be derived from it?

```ts
function getSelectedRouteBadge(selectedRoute, fastestRoute, cheapestRoute) {
  if (selectedRoute === fastestRoute.name) {
    return 'fastest';
  } else if (selectedRoute === cheapestRoute.name) {
    return 'cheapest';
  } else {
    return '';
  }
}

```
Likewise, maybe even for highLighted route, because aren't they supposed to be one and the same which is what this effect does on line 101?